src/progress.rs

```diff
@@ -1080,19 +1081,25 @@ impl MultiProgress {
     /// object overriding custom `ProgressDrawTarget` settings.
     pub fn add(&self, pb: ProgressBar) -> ProgressBar {
         let mut state = self.state.write().unwrap();
-        let object_id = self.id_counter.fetch_add(1, Ordering::SeqCst);
-        state.objects.insert(
-            object_id,
-            MultiObject {
+        let object_idx;
+        if let Some(i) = state.objects.iter().position(|x| x.is_none()) {
```

Comment:
I think I suggested a free list in my previous commit, did you miss that?

Reply:
I tried, however I was unsure about the code complexity added. Given your suggestion I've made the changes in the followup commit.